Makefile

```diff
@@ -0,0 +1,39 @@
+RESET_TARGET := upstream/master
+override PKGS := $(shell go list ./... | grep -v /vendor/)
+override BRANCH := $(shell sh -c "git branch | sed -n 's|* \(.*\)|\1|p'")
+override STAGED_GO_FILES := $(shell git diff --name-only --cached | grep -E ".*.go")
+
+.PHONY: squash format fmt dep test docker/test
+
+help:
+	@grep -E '^[a-zA-Z_-]+:.*?## .*$$' $(MAKEFILE_LIST) | sort | awk 'BEGIN {FS = ":.*?## "}; {printf "\033[36m%-30s\033[0m %s\n", $$1, $$2}'
```

Comment:
A comment would be useful here. Guessing it prints every target including the comment in this file, but oh nelly it's a beast.

Reply:
Yeah, I pulled it from online since it's pretty common procedure with self documentation make files to avoid having to constantly update some help table

Reply:
Will add a comment

Reply:
that's an awesome one-liner : )